### PR TITLE
docs: documentation of rename_XXX

### PR DESCRIPTION
**Pull request by rokamu623**
Signed-off-by: rokamu623 <r.okamura.061@ms.saitama-u.ac.jp>

## Description
add docstring for Architecture::rename_XXX function

## Related links
https://github.com/tier4/CARET_analyze/pull/156

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/architecture/architecture.py [hunk 1]
```diff
@@ -377,13 +377,35 @@ def assign_variable_passings(self, node_name: str,
...
     def rename_callback(self, src: str, dst: str) -> None:
+        """
+        Update callback name "src" to "dst" in architecture.
+
+        Parameters
+        ----------
+        src : str
+            callback name to be updated
+        dst : str
+            callback name to be updating
```
Comment (keita1523) [inline src/caret_analyze/architecture/architecture.py:388]: These may be a nuance which is difficult to interpret.
```suggestion
        src : str
            current callback name
        dst : str
            updated callback name
```

Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Applied in [96d7b4e](https://github.com/tier4/CARET_analyze/pull/253/commits/96d7b4e912b35325087313cbd627b4d4ad8ba4be).
Thank you for suggestion!

Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: @keita1523 
Applied in [96d7b4e](https://github.com/tier4/CARET_analyze/pull/253/commits/96d7b4e912b35325087313cbd627b4d4ad8ba4be).
Thank you for suggestion!

**Comment by rokamu623** [comment]
@takam5f2 
A review by Emb member has been completed.
Please confirm.

File: src/caret_analyze/architecture/architecture.py [hunk 2]
```diff
@@ -394,14 +416,47 @@ def rename_node(self, src: str, dst: str) -> None:
...
+        dst : str
+            updated executor name
+
+        """
         e: ExecutorStruct = Util.find_similar_one(src, self._executors, lambda x: x.executor_name)
         e.executor_name = dst
 
     def rename_topic(self, src: str, dst: str) -> None:
+        """
+        Update topic name "src" to "dst" in architecture.
```
[the remark this answers is not in the document]
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fixed it in [8fafabb](https://github.com/tier4/CARET_analyze/pull/253/commits/8fafabbe19183911f845a9b9736d30e1ebf9c51b).

File: src/caret_analyze/architecture/architecture.py [hunk 3]
```diff
@@ -394,14 +416,47 @@ def rename_node(self, src: str, dst: str) -> None:
...
+        dst : str
+            updated path name
+
+        """
         p: PathStruct = Util.find_similar_one(src, self._paths, lambda x: x.path_name)
         p.path_name = dst
 
     def rename_executor(self, src: str, dst: str) -> None:
+        """
+        Update executor name "src" to "dst" in architecture.
```
[the remark this answers is not in the document]
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fixed it in [8fafabb](https://github.com/tier4/CARET_analyze/pull/253/commits/8fafabbe19183911f845a9b9736d30e1ebf9c51b).

File: src/caret_analyze/architecture/architecture.py [hunk 4]
```diff
@@ -394,14 +416,47 @@ def rename_node(self, src: str, dst: str) -> None:
             c.rename_node(src, dst)
 
     def rename_path(self, src: str, dst: str) -> None:
+        """
+        Update path name "src" to "dst" in architecture.
```
[the remark this answers is not in the document]
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fixed it in [8fafabb](https://github.com/tier4/CARET_analyze/pull/253/commits/8fafabbe19183911f845a9b9736d30e1ebf9c51b).

File: src/caret_analyze/architecture/architecture.py [hunk 5]
```diff
@@ -377,13 +377,35 @@ def assign_variable_passings(self, node_name: str,
...
+        """
         cb_s: List[CallbackStruct] =\
             Util.flatten(cb_g.callbacks for cb_g in
                          Util.flatten([e.callback_groups for e in self._executors]))
         c: CallbackStruct = Util.find_similar_one(src, cb_s, lambda x: x.callback_name)
         c.callback_name = dst
 
     def rename_node(self, src: str, dst: str) -> None:
+        """
+        Update node name "src" to "dst" in architecture.
```
[the remark this answers is not in the document]
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fixed it in [8fafabb](https://github.com/tier4/CARET_analyze/pull/253/commits/8fafabbe19183911f845a9b9736d30e1ebf9c51b).

File: src/caret_analyze/architecture/architecture.py [hunk 6]
```diff
@@ -377,13 +377,35 @@ def assign_variable_passings(self, node_name: str,
                               AssignContextReader(node)))
 
     def rename_callback(self, src: str, dst: str) -> None:
+        """
+        Update callback name "src" to "dst" in architecture.
```
[the remark this answers is not in the document]
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Thank you for your suggestion!
Fixed it in [8fafabb](https://github.com/tier4/CARET_analyze/pull/253/commits/8fafabbe19183911f845a9b9736d30e1ebf9c51b).

**Pull request merged**